### PR TITLE
Fix incorrect DataFrame min max computation with NULL

### LINKED ISSUE #6733: DataFrame DateTime column Min and Max calculation doesn't support NULL
Similar to #6659, but for Max(), Min() and Sum().

New Unit tests for Max and Min:

```
[Fact]
public void TestDateTimeComputations_MaxMin_WithNulls()
{
    var dateTimeColumn = new DateTimeDataFrameColumn("DateTime", new DateTime?[]
        {
            null,
            new DateTime(2022, 1, 1),
            new DateTime(2020, 1, 1),
            new DateTime(2023, 1, 1),
            new DateTime(2021, 1, 1),
            null
        });

         Assert.Equal(new DateTime(2020, 1, 1), dateTimeColumn.Min());
         Assert.Equal(new DateTime(2023, 1, 1), dateTimeColumn.Max());
}

[Fact]
public void TestIntComputations_MaxMin_WithNulls()
{
    var column = new Int32DataFrameColumn("Int", new int?[]
        {
            null,
            2,
            1,
            4,
            3,
            null
        });

    Assert.Equal(1, column.Min());
    Assert.Equal(4, column.Max());
}
```
fail with Messages: 

_Assert.Equal() Failure
Expected: 1
Actual:   0_

and

_Assert.Equal() Failure
Expected: 2020-01-01T00:00:00.0000000
Actual:   0001-01-01T00:00:00.0000000_

Unit Test:

```
[Fact]
public void TestComputations_MaxMin_OnEmptyColumn()
{
    var column = new Int32DataFrameColumn("Int");

    Assert.Null(column.Min());
    Assert.Null(column.Max());
}
```

crashes with 

_System.ArgumentOutOfRangeException : Index was out of range. Must be non-negative and less than the size of the collection. (Parameter 'index')_

New Unit tests for Sum:

```
[Fact]
public void TestIntSum_OnColumnWithNullsOnly()
{
    var column = new Int32DataFrameColumn("Int", new int?[] { null, null });
    Assert.Null(column.Sum());
}

[Fact]
public void TestIntSum_OnEmptyColumn()
{
    var column = new Int32DataFrameColumn("Int");
    Assert.Null(column.Sum());
}
```

crashes with 

_System.ArgumentOutOfRangeException : Index was out of range. Must be non-negative and less than the size of the collection. (Parameter 'index')

### PR DESCRIPTION
**Pull request by asmirnov82**
Fixes #6733

### CODE REVIEW

**Comment by JakeRadMSFT** [comment]
Thanks @asmirnov82! Can you merge this into https://github.com/JakeRadMSFT/machinelearning/tree/u/jakerad/generic-math as well?

File: src/Microsoft.Data.Analysis/DateTimeComputation.cs [hunk 1]
```diff
@@ -189,26 +191,37 @@ public void CumulativeSum(PrimitiveColumnContainer<DateTime> column, IEnumerable
...
-                var buffer = column.Buffers[b];
-                var readOnlySpan = buffer.ReadOnlySpan;
+                var readOnlySpan = column.Buffers[b].ReadOnlySpan;
+                var bitmapSpan = column.NullBitMapBuffers[b].ReadOnlySpan;
                 for (int i = 0; i < readOnlySpan.Length; i++)
                 {
+                    int byteIndex = (int)((uint)i / 8);
+
+                    //Check if bit is not set (value is null) - skip
+                    if (((bitmapSpan[byteIndex] >> (i & 7)) & 1) == 0)
```
[the remark this answers is not in the document]
Comment (JakeRadMSFT) [reply src/Microsoft.Data.Analysis/DateTimeComputation.cs:208]: There is an internal or private function that basically does just this.

[the remark this answers is not in the document]
Comment (JakeRadMSFT) [reply src/Microsoft.Data.Analysis/DateTimeComputation.cs:208]: Hmm not finding the one I thought existed ... one second

[the remark this answers is not in the document]
Comment (JakeRadMSFT) [reply src/Microsoft.Data.Analysis/DateTimeComputation.cs:208]: https://github.com/dotnet/machinelearning/pull/6661/files#diff-bdbb32ff7a16375aa92518a41c65d77328cea77a932df1d313c77911b83b5bbcR284

**Review by JakeRadMSFT — CHANGES_REQUESTED** [review]
I'd like the null check to stay consistent everywhere. We can analyze perf later ... and see if in-lining would help.

**Comment by asmirnov82** [comment]
> 

This wasn't fixed by the latest changes in generic-math branch. I merged my changes, however this requires additional work to change new generic implementation (CalculateReduction method), until changes are done 2 new unit tests fail (TestIntComputations_MaxMin_WithNulls and  TestIntComputations_MaxMin_OnEmptyColumn)

**Comment by JakeRadMSFT** [comment]
> > Thanks @asmirnov82! Can you merge this into https://github.com/JakeRadMSFT/machinelearning/tree/u/jakerad/generic-math as well?
> > It's possible this was fixed by other changes in the generic math branch. Can you make sure it meets your needs/add/update tests?
> 
> This wasn't fixed by the latest changes in generic-math branch. I merged my changes, however this requires additional work to change new generic implementation (CalculateReduction method), until changes are done 2 new unit tests fail (TestIntComputations_MaxMin_WithNulls and TestIntComputations_MaxMin_OnEmptyColumn)

Awesome @asmirnov82  - I've made those changes and merged. Can you take a look at them when you have a chance?

If you're up for it -  I'd love for you to review my generic math PR once I have the .NET 8 branch setup. 

It should be ready for review after next week.

**Comment by JakeRadMSFT** [comment]
Merged into Generic Math branch!

**Review by JakeRadMSFT — APPROVED** [review]

**Review by JakeRadMSFT — APPROVED** [review]

**Pull request merged**